### PR TITLE
hacker-guide: add making-a-release.rst

### PR DESCRIPTION
**Pull request by cgay**
Comments welcome...

### CODE REVIEW

File: documentation/hacker-guide/source/topics/making-a-release.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
+******************
+Release Check-list
+******************
+
+There are a number of manual steps involved in making an Open Dylan release and
+it's easy to forget one or two. Hopefully we can automate some of these but for
+now here is a manual check-list.
+
+#. Let people know you plan to make a release. Use the Gitter channel for this.
+   Because the Dylan ncommunity is small there's no need to make a release
```
Comment (pedro-w) [inline documentation/hacker-guide/source/topics/making-a-release.rst]: Typo - 'ncommunity'

File: documentation/hacker-guide/source/topics/making-a-release.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+   release number during that year.  m is generally 0 but could be 1 if a quick
+   patch release were necessary. Note that these tags are used by the Dylan
+   package manager.
+
+#. Build the binaries for supported platforms
+
+#. Update other packages
+
+   * archlinux -- https://aur.archlinux.org/opendylan.git
+   * debian
```
Comment (pedro-w) [inline documentation/hacker-guide/source/topics/making-a-release.rst]: Is there a homebrew recipe too?

File: documentation/hacker-guide/source/topics/making-a-release.rst [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+   Because the Dylan ncommunity is small there's no need to make a release
+   branch; people can just be careful what they commit until the release is
+   out.
+
+#. Test on supported platforms.
+
+   * Do a 3-stage boostrap.
+   * Run test suites and try to figure out if anything serious is broken.
+     (This is an art, currently, and we should make all our tests pass.)
+   * Do some ad-hoc testing.  (Anything specific to mention here?)
```
Comment (pedro-w) [inline documentation/hacker-guide/source/topics/making-a-release.rst]: Windows is different, I think only @housel knows how to do it.
Also do we need to check what version you have to have installed to bootstrap, e.g. will you be able to bootstrap 2019.2 if you have 2014.1 as your current Dylan?

File: documentation/hacker-guide/source/topics/making-a-release.rst [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+   * In the release-info library
+   * In the build/packages/unix/README file
+   * Do a `git grep` for the previous release number, e.g., "2019.1" and see
+     if anything else needs to be updated.
+
+#. Update the release notes. Hopefully these have been maintained as changes
+   were made.  It may be worth scanning the commit logs.
+
+#. Tag the release
```
Comment (housel) [inline documentation/hacker-guide/source/topics/making-a-release.rst]: The procedure I used this time made use of the GitHub Releases tab (https://github.com/dylan-lang/opendylan/releases). I pushed the **Draft a new release** button, uploaded the binaries to that page, then hit **Publish** which automatically tagged master.

File: documentation/hacker-guide/source/topics/making-a-release.rst (hunk 2 above)
Comment (cgay) [reply documentation/hacker-guide/source/topics/making-a-release.rst]: I just added "homebrew" to the list for now.

File: documentation/hacker-guide/source/topics/making-a-release.rst (hunk 1 above)
Comment (cgay) [reply documentation/hacker-guide/source/topics/making-a-release.rst]: done

File: documentation/hacker-guide/source/topics/making-a-release.rst (hunk 3 above)
Comment (cgay) [reply documentation/hacker-guide/source/topics/making-a-release.rst]: Yeah, I think I remember this exact problem happening in the past. i.e., the latest version wasn't usable for bootstrapping on a specific platform.  How does it look now?

File: documentation/hacker-guide/source/topics/making-a-release.rst (hunk 4 above)
Comment (cgay) [reply documentation/hacker-guide/source/topics/making-a-release.rst]: I wrote it up slightly differently: make a draft release, checkout the tag and do a build, upload, publish the release.

**Pull request merged**